src/server.rs

```diff
@@ -84,10 +90,10 @@ pub async fn run(
     // with rustls, the hyper and tower crates [what axum is built on] had to
     // be used directly
 
-    let shared_state = Arc::new(ServerState { orchestrator });
+    let shared_state = Arc::new(ServerState::new(orchestrator));
```

Comment:
Hey @pscoro, just for my own understanding, what's the need of the `new()` method here instead of what we had?

Reply:
This was just a small but relevant improvement, since we are using server state in multiple places. Dont have much additional reasoning for this change.